examples/mnist/mnist_lib_test.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tests for flax.examples.mnist.mnist_lib."""
+
+import itertools
+import os
+import tempfile
+
+from absl.testing import absltest
+
+import jax
+from jax import numpy as jnp
+
+import numpy as onp
+
+from tensorboard.backend.event_processing import directory_watcher
+from tensorboard.backend.event_processing import event_file_loader
+from tensorboard.util import tensor_util
+import tensorflow_datasets as tfds
+
+import mnist_lib
+
+
+# TODO(#290): Refactor logic in testing/benchmark.py to create a
+# utility class for parsing event files and extracting scalar summaries.
+def process_event(event):
+  for value in event.summary.value:
+    yield value
+
+
+def parse_and_return_summary_values(path):
```

Comment:
Can we avoid having this function and instead of parsing the tensorboard log files, just look at the return value from the train functions?

Reply:
Actually no need to return anything, because we can't reliably test the metrics anyways. Let's just test that nothing breaks.